### PR TITLE
Avoid a few avenues for spurious failures

### PR DESCRIPTION
**Pull request by oli-obk**
see https://github.com/rust-lang/miri/pull/3008 for some debugging around this

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -649,6 +648,21 @@ impl dyn TestStatus {
     }
 }
 
+fn retry_with_backoff<T>(mut f: impl FnMut() -> io::Result<T>) -> io::Result<T> {
+    let mut res = f();
+    for i in 0..10 {
+        if let Err(err) = &res {
+            if let io::ErrorKind::WouldBlock = err.kind() {
+                std::thread::sleep(Duration::from_millis(100 * i));
```
Comment (RalfJung) [inline src/lib.rs:656]:
```suggestion
                std::thread::sleep(Duration::from_millis(100 * i + 10));
```
Let's sleep a bit also after the very first attempt?

Comment (oli-obk) [reply src/lib.rs]: oh 😆 good point

File: src/lib.rs [hunk 2]
```diff
@@ -649,6 +648,21 @@ impl dyn TestStatus {
     }
 }
 
+fn retry_with_backoff<T>(mut f: impl FnMut() -> io::Result<T>) -> io::Result<T> {
+    let mut res = f();
+    for i in 0..10 {
+        if let Err(err) = &res {
+            if io::ErrorKind::WouldBlock == err.kind() || i < 5 {
```
Comment (RalfJung) [inline src/lib.rs:655]: Do you really want to retry on arbitrary failures? That sounds wrong to me.

Comment (oli-obk) [reply src/lib.rs:655]: Well... We also get arbitrary permission denied failures on process spawn, and I have no clue why

**Pull request merged**